ydb/core/tx/schemeshard/schemeshard__op_traits.h

```diff
@@ -290,4 +299,65 @@ struct TSchemeTxTraits<NKikimrSchemeOp::EOperationType::ESchemeOpCreateBackupCol
     constexpr inline static bool CreateDirsFromName = true;
 };
 
+template <>
+struct TSchemeTxTraits<NKikimrSchemeOp::EOperationType::ESchemeOpBackupBackupCollection> : public TSchemeTxTraitsFallback {
+    static std::optional<THashMap<TString, THashSet<TString>>> GetRequiredPaths(const TTxTransaction& tx, const TOperationContext& context) {
+        THashMap<TString, THashSet<TString>> paths;
+
+        const auto& backupOp = tx.GetBackupBackupCollection();
+
+        const auto& targetDir = backupOp.GetTargetDir();
+        const TString& targetPath = JoinPath({tx.GetWorkingDir(), tx.GetBackupBackupCollection().GetName()});
+
+        const TPath& bcPath = TPath::Resolve(targetPath, context.SS);
+        {
+            auto checks = bcPath.Check();
+            checks
+                .NotEmpty()
+                .NotUnderDomainUpgrade()
+                .IsAtLocalSchemeShard()
+                .IsResolved()
+                .NotUnderDeleting()
+                .NotUnderOperation()
+                .IsBackupCollection();
+
+            if (!checks) {
+                return {};
+            }
+        }
+
+        Y_ABORT_UNLESS(context.SS->BackupCollections.contains(bcPath->PathId));
+        const auto& bc = context.SS->BackupCollections[bcPath->PathId];
+
+        auto& collectionPaths = paths[targetPath];
+
+        size_t prefixLen = tx.GetWorkingDir().size() + 1;
+
+        for (const auto& item : bc->Description.GetExplicitEntryList().GetEntries()) {
+            Y_ABORT_UNLESS(prefixLen <= item.GetPath().length());
+            TString itemName = item.GetPath().substr(prefixLen, item.GetPath().size() - prefixLen);
+            auto pos = itemName.rfind("/");
+            if (pos != std::string::npos) {
+                itemName.resize(pos);
+                collectionPaths.emplace(JoinPath({targetDir, itemName}));
+            }
+        }
```

Comment:
Paths work here seems to be brittle. Is it possible to express it via splits and joins?